### PR TITLE
writebatch support clear and len API

### PR DESCRIPTION
**Pull request by siddontang**
@BusyJay @zhangjinpeng1987

### CODE REVIEW

File: src/rocksdb.rs [hunk 1]
```diff
@@ -891,6 +891,20 @@ impl WriteBatch {
     pub fn is_empty(&self) -> bool {
         self.count() == 0
     }
+
+    pub fn len(&self) -> usize {
+        unsafe {
+            let mut length: usize = 0;
+            let _ = rocksdb_ffi::rocksdb_writebatch_data(self.inner, &mut length);
```
Comment (BusyJay) [inline src/rocksdb.rs]: Why ignore return value?

File: src/rocksdb.rs [hunk 2]
```diff
@@ -1102,6 +1116,13 @@ mod test {
         let p = db.write(batch);
         assert!(p.is_ok());
         assert!(db.get(b"k1").unwrap().is_none());
+
+        let batch = WriteBatch::new();
+        let prev_len = batch.len();
+        let _ = batch.delete(b"k1");
+        assert!(batch.len() > 0);
+        batch.clear();
+        assert_eq!(batch.len(), prev_len);
```
Comment (BusyJay) [inline src/rocksdb.rs]: Why not use 0 directly?

Comment (siddontang) [inline src/rocksdb.rs]: because the empty writebatch size is not 0.

File: src/rocksdb.rs [hunk 3]
```diff
@@ -891,6 +891,20 @@ impl WriteBatch {
     pub fn is_empty(&self) -> bool {
         self.count() == 0
     }
+
+    pub fn len(&self) -> usize {
```
Comment (BusyJay) [inline src/rocksdb.rs]: Calling it `len` is very confusing. We usually consider it as the number of elements in `WriteBatch` just like `Vec::len`. But here it seems to indicate the memory size of inner data.

File: src/rocksdb.rs (hunk 2 above)
Comment (BusyJay) [inline src/rocksdb.rs]: Then L1123 is useless.

**Comment by BusyJay** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**